Tests/UnitTests/Purchasing/StoreKitAbstractions/SubscriptionPeriodTests.swift

```diff
@@ -45,62 +45,92 @@ class SubscriptionPeriodTests: TestCase {
 
     // Note: can't test creation from `StoreKit.Product.SubscriptionPeriod` because it has no public constructors.
 
+    func testPricePerWeek() {
+        let tests: [Test] = [
+            .init(p(1, .day), 1, 7),
+            .init(p(14, .day), 2, 1),
+            .init(p(1, .week), 10, 10),
+            .init(p(2, .week), 10, 5),
+            .init(p(1, .month), 14.99, 3.74),
+            .init(p(2, .month), 30, 3.75),
+            .init(p(3, .month), 40, 3.33),
+            .init(p(1, .year), 120, 2.3),
+            .init(p(1, .year), 50, 0.95),
+            .init(p(1, .year), 29.99, 0.57),
+            .init(p(3, .year), 720, 4.6)
+        ]
+
+        for test in tests {
+            let pricePerWeek = test.period.pricePerWeek(withTotalPrice: test.price) as NSDecimalNumber
+            let result = Double(truncating: pricePerWeek)
+            let expected = Double(truncating: test.expected as NSDecimalNumber)
+
+            expect(
+                line: test.line,
+                result
+            ).to(beCloseTo(expected),
+                 description: "\(test.price) / \(test.period.debugDescription)")
+        }
+    }
+
     func testPricePerMonth() {
-        let expectations: [(period: SubscriptionPeriod, price: Decimal, expected: Decimal)] = [
-            (p(1, .day), 2, 60),
-            (p(15, .day), 5, 10),
-            (p(1, .week), 10, 40),
-            (p(2, .week), 10, 20),
-            (p(1, .month), 14.99, 14.99),
-            (p(2, .month), 30, 15),
-            (p(3, .month), 40, 13.33),
-            (p(1, .year), 120, 10),
-            (p(1, .year), 50, 4.16),
-            (p(1, .year), 29.99, 2.49),
-            (p(3, .year), 720, 20)
+        let tests: [Test] = [
+            .init(p(1, .day), 2, 60),
```

Comment:
This makes failures come from each of these lines who makes it much easier to debug.